### PR TITLE
feat(trading): add market banner for successor markets

### LINKED ISSUE #4133: There is a successor market to this market
## Story

As a user
I want to know if there is a successor to this market and whether there is trading volume there
So that I know to move there / consider it based on my current position

## Acceptance Criteria

- [x] In market view, I can see this market is nearing trading termination and has a successor
- [x] I can get to the successor market from the parent market i.e. there is a link 
- [x] I can see the trading volume of the successor market 
- [x] I can see when this current market (parent market) is due to terminate 
- [x] On the successor market, there is a way to see the parent market ID (in info tab)

## Tasks

- [ ] UX (if needed)
- [ ] Design (if needed)
- [ ] Team and stakeholder review
- [ ] Specs reviewed and created or adjusted
- [x] Implementation
- [ ] Testing (unit and/or e2e)
- [ ] Code review
- [ ] QA review

## Sketch
X Platform UX - There is a successor market to this market

### PR DESCRIPTION
**Pull request by macqbat**
MERGE AFTER Core DEPLOYMENT!
# Related issues 🔗

Closes #4133 

# Description ℹ️

Display banner with message informs about there is successor of current market.
Display parent market id of current market in market info accordion

# Demo 📺

Screenshot 2023-07-06 at 10 14 14
Screenshot 2023-07-06 at 10 17 04
Screenshot 2023-07-06 at 10 19 39
Screenshot 2023-07-06 at 10 36 56

# Technical 👨‍🔧

Details of technical implementation that reviewers may need to be aware of, if applicable.

### CODE REVIEW

File: apps/trading/components/banner/market-successor-banner.tsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+const getExpiryDate = (tags: string[], close?: string): Date | null => {
+  const expiryDate = getMarketExpiryDate(tags);
+  return expiryDate || (close && new Date(close)) || null;
+};
+
+export const MarketSuccessorBanner = () => {
+  const { marketId } = useParams();
+  const { pathname } = useLocation();
+  const isMarketPage = pathname.match(/^\/markets\/(?!(all)).+/);
```
Comment (bglownia) [inline apps/trading/components/banner/market-successor-banner.tsx:31]: We could avoid of rendering by using router somewhere above

```
let element = useRoutes([
    {
      path: "/markets/:marketId",
      element: () => {
        const { marketId } = useParams();
        return marketId  === 'all' ? null : <MarketSuccessorBanner marketId={marketId} />
      },      
  ]);

  return element;
```

Comment (macqbat) [reply apps/trading/components/banner/market-successor-banner.tsx:31]: it's a little bit outside of scope of this issue

**Review by pscott31 — APPROVED** [review]

File: apps/trading/components/banner/market-successor-banner.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  const expiryDate = getMarketExpiryDate(tags);
+  return expiryDate || (close && new Date(close)) || null;
+};
+
+export const MarketSuccessorBanner = () => {
+  const { pathname } = useLocation();
+  const isMarketPage = pathname.match(/^\/markets\/(?!(all)).+/);
+  const marketId = isMarketPage ? (pathname.split('/').pop() as string) : '';
+
+  const [dismissed, setDismissedInStorage] = useLocalStorageSnapshot(
```
Comment (mattrussell36) [inline apps/trading/components/banner/market-successor-banner.tsx:34]: When did we agree that the banner should be dismiss-able and for that state to be stored? I find this unnecessary. If a market has been succeeded then in almost all cases the market will have terminated. Can we remove this functionality?

File: apps/trading/pages/_app.page.tsx [hunk 3]
```diff
@@ -99,6 +103,7 @@ function AppBody({ Component }: AppProps) {
           />
           <ViewingBanner />
           <UpgradeBanner showVersionChange={true} />
+          <MarketSuccessorBanner />
```
Comment (mattrussell36) [inline apps/trading/pages/_app.page.tsx:106]: We should move this banner so it is displayed within the market page, then we dont need the path matching logic

**Review by mattrussell36 — CHANGES_REQUESTED** [review]

File: apps/trading/components/banner/market-successor-banner.tsx (hunk 2 above)
Comment (mattrussell36) [reply apps/trading/components/banner/market-successor-banner.tsx:34]: I'm wrong on "all cases the market will have terminated". This is not true, there can be overlap between the parent and the successor. Nonetheless I'd like to not make the banner dismissable for this first iteration

Comment (macqbat) [reply apps/trading/components/banner/market-successor-banner.tsx]: I removed functionality of storing the state

File: apps/trading/client-pages/market/market.tsx [hunk 4]
```diff
@@ -132,7 +133,10 @@ export const MarketPage = () => {
         marketName={data?.tradableInstrument.instrument.name}
         decimalPlaces={data?.decimalPlaces}
       />
-      {tradeView}
+      <div className="flex flex-col h-full">
+        <MarketSuccessorBanner />
```
Comment (mattrussell36) [inline apps/trading/client-pages/market/market.tsx:137]: We should keep market specific banners together. Can you move this to the trade grid and locate it next to the oracle banner?

Then we can also just pass market data as a prop which would simplify the banner logic a bit.

```
<div className="col-span-2 bg-vega-green">
        <MarketSuccessorBanner market={market} />
        <OracleBanner marketId={market?.id || ''} />
</div>
```

**Review by mattrussell36 — CHANGES_REQUESTED** [review]

File: apps/trading/client-pages/market/market.tsx (hunk 4 above)
Comment (macqbat) [reply apps/trading/client-pages/market/market.tsx]: done

**Review by mattrussell36 — APPROVED** [review]

**Pull request merged**